versioning/BRM/extentmap.h

```diff
@@ -1037,109 +1028,140 @@ class ExtentMap : public Undoable
-                                        uint32_t  partitionNum,
-                                        uint16_t  segmentNum,
-                                        execplan::CalpontSystemCatalog::ColDataType colDataType,
-                                        uint32_t& startBlockOffset);
-    LBID_t _createDictStoreExtent(uint32_t size, int OID,
-                                  uint16_t  dbRoot,
-                                  uint32_t  partitionNum,
-                                  uint16_t  segmentNum);
-    bool isValidCPRange(int64_t max, int64_t min, execplan::CalpontSystemCatalog::ColDataType type) const;
-    void deleteExtent(const int emIndex, const bool clearEMIndex = true);
-    LBID_t getLBIDsFromFreeList(uint32_t size);
-    void reserveLBIDRange(LBID_t start, uint8_t size);    // used by load() to allocate pre-existing LBIDs
-
-    key_t chooseEMShmkey();
-    key_t chooseFLShmkey();
-    key_t chooseEMIndexShmkey();
-    key_t getInitialEMIndexShmkey() const;
-    //see the code for how keys are segmented
-    key_t chooseShmkey(const MSTEntry* masterTableEntry, const uint32_t keyRangeBase ) const;
-    void grabEMEntryTable(OPS op);
-    void grabFreeList(OPS op);
-    void grabEMIndex(OPS op);
-    void releaseEMEntryTable(OPS op);
-    void releaseFreeList(OPS op);
-    void releaseEMIndex(OPS op);
-    void growEMShmseg(size_t nrows = 0);
-    void growFLShmseg();
-    void growEMIndexShmseg(const size_t suggestedSize = 0);
-    void finishChanges();
-
-    EXPORT unsigned getFilesPerColumnPartition();
-    unsigned getExtentsPerSegmentFile();
-    unsigned getDbRootCount();
-    void getPmDbRoots(int pm, std::vector<int>& dbRootList);
-    DBRootVec getAllDbRoots();
-    void checkReloadConfig();
-    ShmKeys fShmKeys;
-
-    bool fDebug;
-
-    int _markInvalid(const LBID_t lbid, const execplan::CalpontSystemCatalog::ColDataType colDataType);
-
-    void loadVersion4(std::ifstream& in);
-    void loadVersion4(idbdatafile::IDBDataFile* in);
-
-    ExtentMapImpl* fPExtMapImpl;
-    FreeListImpl* fPFreeListImpl;
-    ExtentMapIndexImpl* fPExtMapIndexImpl_;
+  LBID_t _createDictStoreExtent(uint32_t size, int OID, uint16_t dbRoot, uint32_t partitionNum,
+                                uint16_t segmentNum);
+
+  // Delete extent.
+  ExtentMapRBTree::iterator deleteExtent(ExtentMapRBTree::iterator it, const bool clearEMIndex = true);
+
+  bool isValidCPRange(int64_t max, int64_t min, execplan::CalpontSystemCatalog::ColDataType type) const;
+  LBID_t getLBIDsFromFreeList(uint32_t size);
+  void reserveLBIDRange(LBID_t start, uint8_t size); // used by load() to allocate pre-existing LBIDs
+  std::vector<EMEntry> getEmIdentsByLbids(const std::vector<LBID_t>& lbids);
+  std::vector<ExtentMapRBTree::iterator> getEmIteratorsByLbids(const std::vector<LBID_t>& lbids);
+
+  // Choose keys.
+  key_t chooseEMShmkey();
+  key_t chooseFLShmkey();
+  key_t chooseEMIndexShmkey();
+
+  // TODO: Add comments.
```

Comment:
There are two options: remove the comment or add more comments b/c this TODO will last forever.

Reply:
Sure, thanks.